### PR TITLE
Delete bundle authorizations when users delete their data

### PR DESCRIPTION
**Pull request by suecarmol**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Bundle authorizations were not being deleted when a user deleted their data. This caused authorizations to be without an attached user.

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
A bundle authorization without an attached user is of no use in the database, so it would be best if we deleted it.

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
[T258380](https://phabricator.wikimedia.org/T258380)

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
Added a unit test

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: TWLight/users/views.py [hunk 1]
```diff
@@ -562,6 +562,12 @@ def delete(self, request, *args, **kwargs):
             user_authorization.date_expires = date.today() - timedelta(days=1)
             user_authorization.save()
 
+        # Delete bundle authorizations, if user is eligible for bundle
+        if user.editor.wp_bundle_authorized:
```
Comment (Samwalton9) [inline TWLight/users/views.py:566]: As I understand it, this is a bit of a duplicate check - `wp_bundle_authorized` returns True if a bundle authorization exists (also using `get_bundle_authorization`), and then checks that the authorization `is_valid`. But in this context I don't think we actually care if the auth is valid or not (though it always should be), we'd want to delete it either way. So I think we can just jump straight to seeing if we get an object from `get_bundle_authorization` or not.

File: TWLight/users/views.py [hunk 2]
```diff
@@ -562,6 +562,12 @@ def delete(self, request, *args, **kwargs):
...
+            bundle_auths = user.editor.get_bundle_authorization
+            if bundle_auths is not None:
+                bundle_auths.delete()
+
```
Comment (jsnshrmn) [inline TWLight/users/views.py:570]: Agree with @Samwalton9 , something like this should do just fine.
```suggestion
        # Delete any bundle authorizations.
        bundle_auths = user.editor.get_bundle_authorization
            if bundle_auths:
                bundle_auths.delete()

```

File: TWLight/users/views.py (hunk 1 above)
Comment (suecarmol) [reply TWLight/users/views.py]: I've changed the code to exclude that condition!

File: TWLight/users/views.py (hunk 2 above)
Comment (suecarmol) [reply TWLight/users/views.py]: Done!

**Pull request merged**